app/middlewares/auth.js

```diff
@@ -0,0 +1,36 @@
+const { unauthorizedError } = require('../errors');
+const { verifyToken } = require('../helpers/auth');
+const logger = require('../logger');
+const { objIsNotEmpty } = require('../mappers/commonObjectsValidations');
+const userService = require('../services/users');
+
+exports.validateAccess = async (req, res, next) => {
+  try {
+    const token = req.headers.authorization ? req.headers.authorization.replace(/^Bearer\s+/, '') : '';
+
+    const tokenDecoded = await verifyToken(token);
+
+    if (!tokenDecoded) {
+      return next(unauthorizedError('Unauthorized access'));
+    }
+
+    // eslint-disable-next-line require-atomic-updates
+    req.tokenDecoded = tokenDecoded;
+    return next();
+  } catch (error) {
+    logger.error(`validateAccess Error => ${error}`);
+    return new Error(`validateAccess Error => ${error.message}`);
```

Comment:
I think that in the catch you could throw a error using the standar and return next. For example 

`return next(decodeTokenError('An error was occurred when try to decode token'))`.